File: Day22bst.java
import java.util.*;
import java.io.*;
class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data=data;
        left=right=null;
    }
}
//MY OWN CODE (ONLY SOLUTION CLASS)          ********************************
class Solution{

	public static int getHeight(Node root){
      //Write your code here
      int leftt,rightt;
      if (root==null||(root.left==null && root.right==null))
        return 0;
      else {
        leftt=getHeight(root.left);
        rightt=getHeight(root.right);
        if (leftt>rightt)
            return 1+leftt;
        else 
        return 1+rightt;
      //return (1+Java.lang.Math.max(leftt,rightt));  
      }  
      
    }
//CODE ENDS             ******************************** **********************
    public static Node insert(Node root,int data){
        if(root==null){
            return new Node(data);
        }
        else{
            Node cur;
            if(data<=root.data){
                cur=insert(root.left,data);
                root.left=cur;
            }
            else{
                cur=insert(root.right,data);
                root.right=cur;
            }
            return root;
        }
    }
	 public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int T=sc.nextInt();
        Node root=null;
        while(T-->0){
            int data=sc.nextInt();
            root=insert(root,data);
        }
        int height=getHeight(root);
        System.out.println(height);
    }
}

File: Day22bst.py
class Node:
    def __init__(self,data):
        self.right=self.left=None
        self.data = data
class Solution:
    def insert(self,root,data):
        if root==None:
            return Node(data)
        else:
            if data<=root.data:
                cur=self.insert(root.left,data)
                root.left=cur
            else:
                cur=self.insert(root.right,data)
                root.right=cur
        return root
## MY CODE STARTS HERE(GET HEIGHT FUNCTION#)######################################
    def getHeight(self,root):
        #Write your code here
        leftt=right=0
        if (root==None or (root.left==None and root.right==None)):
            return 0
        else:
            leftt=self.getHeight(root.left)
            rightt=self.getHeight(root.right)
            return 1+max(leftt,rightt)
### MY CODE ENDS HERE                    #############################################            
T=int(raw_input())
myTree=Solution()
root=None
for i in range(T):
    data=int(raw_input())
    root=myTree.insert(root,data)
height=myTree.getHeight(root)
print height      
